### DISCUSSION TITLE
gdscript highlight query is broken

### DISCUSSION THREAD

**Discussion by aekobear** [helix-editor/helix#11755] [Troubleshooting]
### Summary

With the latest `gdscript/highlights.scm`, I no longer have syntax highlighting for gd files. In the logs, I get this error:

> 2024-09-22T10:38:55.179 helix_core::syntax [ERROR] Could not parse queries for language "gdscript". Are your grammars out of sync? Try running 'hx --grammar fetch' and 'hx --grammar build'. This query could not be parsed: QueryError { row: 101, column: 3, offset: 1851, message: "when", kind: NodeType }

If I replace my file `runtime/queries/gdscript/highlights.scm` with the one from the previous commit:

https://github.com/helix-editor/helix/blob/64c137bd3d7ac7f82f538f5dd520e3aa8bca0d1a/runtime/queries/gdscript/highlights.scm

Highlighting works properly again

Note that I did try the following:

- copying the latest runtime from the repo
- running `hx --grammar fetch` and `hx --grammar build`

Neither worked

### Reproduction Steps

 

I tried this:

1. make `file.gd` with contents:

```gdscript
extends Node

func test():
  pass
  ```
  
  2. hx file.gd

I expected this to happen:

- Syntax highlighting

Instead, this happened:

- Query parsing error for gdscript

### Helix log

I ran `hx -vv file.gd` and this was the only relevant output

<details><summary>~/.cache/helix/helix.log</summary>

```
2024-09-22T10:38:55.179 helix_core::syntax [ERROR] Could not parse queries for language "gdscript". Are your grammars out of sync? Try running 'hx --grammar fetch' and 'hx --grammar build'. This query could not be parsed: QueryError { row: 101, column: 3, offset: 1851, message: "when", kind: NodeType }
```

</details>

### Platform

Linux

### Terminal Emulator

COSMIC terminal

### Installation Method

releases page (linux appimage)

### Helix Version

helix 24.7 (079f5442)

**Comment by the-mikedavis** [comment]
This is an issue with your environment rather than the queries. That log message means that the version of the gdscript tree-sitter parser and the queries is out of sync: the queries match on a "when" type that the version of the parser you're using doesn't expose. You can't copy the latest query file from this repository: the query file needs to be compatible with the parser version you have. `hx -g fetch` and `hx -g build` compiles the parser at the version in the `languages.toml` in the Helix repository rather than the latest so if you want to use the most recent highlights.scm file you will also need to update your Helix installation.

**Reply by aekobear** [reply]
Thank you! Looking into it further, I think my issue stems from the fact that I am using the appimage version of helix which does not need a runtime folder at all, but I still had a runtime folder from a previous installation. I deleted the runtime folder entirely and now everything is working as expected
